src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -48,7 +49,13 @@ def validate(self, attrs):
         teamnames, usernames, emails = parse_code_owners(attrs["raw"])
 
         # Check if there exists Sentry users with the emails listed in CODEOWNERS
-        user_emails = UserEmail.objects.filter(email__in=emails)
+        user_emails = UserEmail.objects.annotate(
+            user_in_org=Exists(
+                OrganizationMember.objects.filter(
+                    user=OuterRef("user"), organization=self.context["project"].organization
+                )
+            )
+        ).filter(email__in=emails, user_in_org=True)
```

Comment:
You can write this as:

`UserEmail.objects.filter(user__sentry_orgmember_set__organization=self.context["project"].organization)`

Reply:
oop didn't see this, @wedamija just out of curiosity, is how it's currently written just doing the same thing?

Reply:
It's a pretty different query, the original does:
```
SELECT "sentry_useremail"."id", "sentry_useremail"."user_id", "sentry_useremail"."email", "sentry_useremail"."validation_hash", "sentry_useremail"."date_hash_added", "sentry_useremail"."is_verified", 
EXISTS(
    SELECT U0."id", U0."organization_id", U0."user_id", U0."email", U0."role", U0."flags", U0."token", U0."date_added", 
    U0."token_expires_at", U0."has_global_access", U0."inviter_id", U0."invite_status", U0."type" 
    FROM "sentry_organizationmember" U0 WHERE (
        U0."user_id" = ("sentry_useremail"."user_id") 
        AND U0."organization_id" = 1
    )
) AS "user_in_org" 
FROM "sentry_useremail" 
WHERE (
    "sentry_useremail"."email" IN (hi@hi.com) 
    AND EXISTS(
        SELECT U0."id", U0."organization_id", U0."user_id", U0."email", U0."role", U0."flags", U0."token", U0."date_added", 
        U0."token_expires_at", U0."has_global_access", U0."inviter_id", U0."invite_status", U0."type" 
        FROM "sentry_organizationmember" U0 
        WHERE (
            U0."user_id" = ("sentry_useremail"."user_id") 
            AND U0."organization_id" = 1
        )
    ) = True
)
```

My suggestion does:
```
SELECT "sentry_useremail"."id", "sentry_useremail"."user_id", "sentry_useremail"."email", "sentry_useremail"."validation_hash", 
"sentry_useremail"."date_hash_added", "sentry_useremail"."is_verified" 
FROM "sentry_useremail" 
INNER JOIN "auth_user" ON ("sentry_useremail"."user_id" = "auth_user"."id") 
INNER JOIN "sentry_organizationmember" ON ("auth_user"."id" = "sentry_organizationmember"."user_id") 
WHERE "sentry_organizationmember"."organization_id" = 1
```

The main difference here is mine uses `JOIN` where the other uses `EXISTS`. Using joins can lead to duplicate rows in some cases, although in this case we should be fine. Both queries work fine, and if I were writing pure SQL I'd rather use `EXISTS`, but I don't like how in our current version of Django it includes the exist statement in the `SELECT` for no reason.

Reply:
@wedamija 👍 yeah I know what I want in SQL but not the best way to implement it with the Django ORM.